contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -79,12 +82,17 @@ def register_options(cls, register):
              help='Prevent test failure but still produce output for problems.')
     register('--interpreter-constraints-whitelist', fingerprint=True,
              default=None,
+             removal_version='1.14.0.dev2',
```

Comment:
If there's no rush as you allude below, it would be good to stick to our documented policy and make this 1.16.0.dev2.

Reply:
What's the hint - as noted it probably makes sense to pull Chris in and make sure the deprecation story is sorted and explained rationally in the user warning messaging.